core/src/main/java/de/javakaffee/web/msm/MemcachedSessionService.java

```diff
@@ -696,7 +696,13 @@ public MemcachedBackupSession createSession( String sessionId ) {
         }
 
         _manager.incrementSessionCounter();
-
+        //if the new session exist in _invalidSessionsCache, we should remove it marking this session valid.(#284)
+        if( _invalidSessionsCache.get(session.getId()) ){
```

Comment:
The jvm has to unbox the Boolean, therefore this would throw an NPE if there's no entry for the given session id. Just use containsKey instead of get.